### ISSUE TITLE
EarlyStopping is not early stopping

### ISSUE THREAD

**Issue by bartolsthoorn** [atomistic-machine-learning/SchNet#3]
I am new to Tensorflow so I could be wrong, but is it correct to say that the `EarlyStopping` implemented in this repo is confusing?

The only stop condition I can find is `if step > max_steps:`, whereas "early stopping" seems to suggest the training is stopped when the training or validation loss is no longer declining.
